provider/namespace_type_filter.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	sd "github.com/aws/aws-sdk-go/service/servicediscovery"
+)
+
+const (
+	namespaceTypePublic  = "public"
+	namespaceTypePrivate = "private"
+)
+
+// NamespaceTypeFilter holds a namespace type to filter for.
+type NamespaceTypeFilter struct {
```

Comment:
these types and above const should be prefixed with AWS

Reply:
Will be fixed 👍